### DISCUSSION TITLE
tildes in OBX-5

### DISCUSSION THREAD

**Discussion by Bugzy-can** [nextgenhealthcare/connect#5821] [General]
A tilde in a standard HL7 message represents a carriage return, so if an OBX is sent like

"OBX|1|TX|||This is a sentence~~This is more information~and it continues|||||||||||||||||||||||||||||||"

It should be handled the same as

OBX|1|TX|||This is a sentence|||||||||||||||||||||||||||||||
OBX|2|TX||||||||||||||||||||||||||||||||||
OBX|3|TX||||||||||||||||||||||||||||||||||
OBX|4|TX|||This is more information|||||||||||||||||||||||||||||||
OBX|5|TX|||and it continues|||||||||||||||||||||||||||||||

Unfortunately the conversion from text (incoming HL7) to XML, changes tildes into <OBX.5.1>This is a sentence</OBX.5.1> etc.

This makes a mess of the report.

In preprocessor I can clean this up using

```
var myArray = message.split("|");
var workingStr = "";
for (var cnt = 0; cnt < myArray.length; cnt++ )
{
   if (myArray[cnt].includes("OBX", 0))
   {   	
      //Found OBX
      workingStr = myArray[cnt + 5];
      workingStr = workingStr.replace(/~/g, "< br >")
      myArray[cnt + 5] = workingStr;      
   }
}
return myArray.join("|");
```
Or alternately, I can clean up the mess in the Destination Transformer using replace commands, but I am wondering if there may be a simpler way to fix this?

**Comment by pacmano1** [comment]
Please revise your post and use code blocks.   See https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by pacmano1** [reply]
Please also visit the forums if you have not already, see https://forums.mirthproject.io/forum/mirth-connect/support/183380-replacing-tilde-in-obx-5-body-of-result-report-with-space for more example code, there are more threads over there also.  It depends on what you really want to do when you consume that message.  

Also don't forget in mirth you can change this to not consider all line feeds / CRs, etc (after you do the regex) as a segment delimeter:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/88761ba1-1887-4f20-91a1-e004b620fee4)

**Comment by Bugzy-can** [comment]
Thanks. I cleaned up the code, and it does look better.

I did look at that exact post in the forums, but the issue is not the same. In that post, they want to strip the tildes from a multi-OBX report. In my issue, this is a single OBX segment that is using tildes where a CR needs to be, so the same thing as multiple OBX, only it comes over in a single OBX. Very common in HL7.
Also in that post, the suggested fix in pre-processor inserts twice the number of CR's. Well, it was a suggestion. Mine works much better. 

Not sure what is highlighted here with Convert Line Breaks. Not seeing how this would affect a tilde? Is there some connection?

**Reply by pacmano1** [reply]
If I understand your objective it is to preserve the line breaks in the report.  It is irrelevant if that is one or more OBX segments.  That forum post (although it converts tildes to "") handles _one or more_ OBX segments.

You can convert the tildes to a \n in the preprocessor and uncheck the option in the screenshot above, that is why I showed it:

Preprocessor:  (note when you post code you can add the language also)

```JS
// Modify the message variable below to pre process data
const result = message.replace(/^OBX.*$/gm, function(match) {
return match.replace(/~/g, "\n");
});

return result;​
```
![image](https://github.com/nextgenhealthcare/connect/assets/44065187/268ff3bd-4ee6-4b1e-811d-852e78a373c6)

After the preprocesor:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/ddad47e0-0249-4740-a877-774c8afb157b)

In source transformer:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/2b5cb739-7b3e-47f0-a729-cf47f783dbcc)

And the logger:

![image](https://github.com/nextgenhealthcare/connect/assets/44065187/e0afa829-be8e-4a64-b78d-8ede4d34ab09)

**Reply by pacmano1** [reply]
So in the end, what is better?  No clue.  I prefer regex for stuff like this.   And that preprocessor statement (taken almost exactly from my post on the forums except changing "" to "\n") is the reason I pointed you there.

**Comment by Bugzy-can** [comment]
Nice, thanks. :)
You are awesome.
